File: Backend/BusinessLogic/Models/OrderModel.cs
﻿using System;
using System.Collections.Generic;

namespace BusinessLogic.Models
{
    public class OrderModel
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int SessionId { get; set; }
        public PriceModel Price { get; set; } = null!;
        public DateTime RegistratedAt { get; set; }
        public SessionSeatsModel SessionSeats { get; set; } = null!;
        public IReadOnlyCollection<CinemaServiceModel> CinemaServices { get; set; } = null!;
        public DateTime SessionStart { get; set; }
        public int CinemaId { get; set; }
        public string CinemaName { get; set; } = null!;
        public int HallId { get; set; }
        public string HallName { get; set; } = null!;
    }
}

File: Frontend/src/app/Models/OrderModel.ts
import { PriceModel } from './PriceModel';

export class OrderModel {
    id : number = 0;
    userId : number = 0;
    sessionId : number = 0;
    price : PriceModel = new PriceModel();
    sessionStart : Date = new Date();
    cinemaId : number = 0;
    cinemaName : string = '';
    hallId : number = 0;
    hallName : string = '';
}